log/ratelimit_test.go

```diff
@@ -0,0 +1,176 @@
+package log
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRateLimitedLoggerLogs(t *testing.T) {
+	buf := bytes.NewBuffer(nil)
+	c := newCounterLogger(buf)
+	reg := prometheus.NewPedanticRegistry()
+	r := NewRateLimitedLogger(c, 1, 1, reg)
+
+	r.Errorln("Error will be logged")
+	assert.Equal(t, 1, c.count)
+
+	logContains := []string{"error", "Error will be logged"}
+	c.verify(t, logContains)
+}
+
+func TestRateLimitedLoggerLimits(t *testing.T) {
+	buf := bytes.NewBuffer(nil)
+	c := newCounterLogger(buf)
+	reg := prometheus.NewPedanticRegistry()
+	r := NewRateLimitedLogger(c, 2, 2, reg)
+
+	r.Errorln("error 1 will be logged")
+	assert.Equal(t, 1, c.count)
+	c.verify(t, []string{"error", "error 1 will be logged"})
+
+	r.Infoln("info 1 will be logged")
+	assert.Equal(t, 2, c.count)
+	c.verify(t, []string{"info", "info 1 will be logged"})
+
+	r.Debugln("debug 1 will be discarded")
+	assert.Equal(t, 2, c.count)
```

Comment:
[nit] To get even an higher confidence, I suggest you to assert also on "not containts". So, check that "debug 1 will be discarded" is **not** in included in the logged lines.